src/types.rs

```diff
@@ -584,6 +588,12 @@ pub enum KeySerialization {
     EcdhEsHkdf256,
     Raw,
     Sec1,
+    /// Used by backends implementing RSA.
+    ///
+    /// Since RSA keys have multiple parts, and that the [`SerializeKey`](crate::api::Reply::SerializeKey) and
+    /// [`UnsafeInjectKey`](crate::api::Request::UnsafeInjectKey) have only transfer one byte array, the RSA key is serialized with postcard
+    RsaParts,
```

Comment:
How about calling this `Serde` or `PostcardSerde` to make it potentially reusable?

Reply:
I want to keep it generic and let the backend decide on the actual data format (for now at least)

Reply:
The serialization format doesn't affect how they're stored. I don't want to reuse `Raw` as it feels misleading. Unlike ECC, there is no standard "raw" byte representation of RSA keys. 

The "Parts" is in contrast to the PKCS format. The PKCS format is defined, but the "Parts" formats contains each element of a public key separately. The "part" name is more representative than "Internal" or "Private". "Serde" feels out of place. The use of postcard and serde is more of a workaround that the actual goal of the API. If you look at the traits in `trussed-rsa-backend`, they provide APIs directly over the structures, and hide away the parsing/serializing with postcard and serde (when possible):

```rust
fn unsafe_inject_rsa4096<'c>(
    &'c mut self,
    key_parts: RsaImportFormat,
    attributes: StorageAttributes,
) -> ClientResult<'c, reply::UnsafeInjectKey, Self>;

fn deserialize_rsa4096_public_key<'c>(
    &'c mut self,
    key_parts: RsaPublicParts,
    attributes: StorageAttributes,
) -> ClientResult<'c, reply::DeserializeKey, Self>;

```

Ideally I would even merge the `RsaImportFormat` an `RsaPublicParts` structures into Trussed proper but I want to be sure we can have something that fits all back-ends and use-cases first.

Maybe we should have mechanism extensions? That would mean more generics though.

Reply:
Yeah I understand. My point is just that `Pkcs8Der` is an obviously valid general purpose format to upstream, while `RsaParts` is not. So can I'd like to merge this without this `RsaParts`. I believe you can use `Raw` (or we add something generic but general-purpose like `Experimental`) for your current work, and once it's set in stone, do a follow-up PR?

Reply:
Or what about `Parts` :) That way you have your specific format to use, but unrelated algorithms might find a use too.

Reply:
> My point is just that `Pkcs8Der` is an obviously valid general purpose format to upstream, while `RsaParts` is not

I disagree. Whatever we do we **will** need a way to access specific parts of RSA keys. Both PIV and OpenPGP need that (OpenPGP needs it for both public and private keys, PIV at least for public keys). `Pkcs8Der` will also be incompatible with the SE050.

> Or what about `Parts` :) That way you have your specific format to use, but unrelated algorithms might find a use too.

I'm not a fan of loosing readability for more flexibility when there are no obvious benefits to it. Do you know of any other algorithm that would use this?